### ISSUE TITLE
Time picker invalid state does not match component spec

### ISSUE THREAD

**Issue by emyarod** [carbon-design-system/carbon#6518]
The time picker invalid state does not match the component spec. The error text should be red, and when the input field is focused, the focus outline should not override the error state outline

originally reported by @laurenmrice 

![Jul-22-2020 10-40-51](https://user-images.githubusercontent.com/8265238/88200625-06901080-cc0c-11ea-8fee-68e3198bb999.gif)
